src/lein_monolith/plugin.clj

```diff
@@ -82,16 +99,70 @@
 (defn build-inherited-profiles
   "Returns a map from profile keys to inherited profile maps."
   [monolith subproject]
-  (let [inherit-profile (inherited-profile
-                          monolith :inherit
-                          (:monolith/inherit subproject))
-        leaky-profile (inherited-profile
-                        monolith :inherit-leaky
-                        (:monolith/leaky subproject (boolean (:monolith/inherit subproject))))]
-    (cond-> nil
-      inherit-profile (assoc :monolith/inherited inherit-profile)
-      leaky-profile (assoc :monolith/leaky (vary-meta leaky-profile assoc :leaky true)))))
+  (reduce
+    (fn [acc [profile-key {:keys [types ks]}]]
+      (let [profile (some-> (choose-inheritance-source monolith types)
+                            (inherited-profile subproject ks)
+                            (maybe-mark-leaky types))]
+        (cond-> acc
+          profile (assoc profile-key profile))))
+    nil
+    profile-config))
+
+
+(def ^:private init-lock
```

Comment:
Hmm, I really don't think that the plugin code should be doing any subproject initialization. That should _only_ be needed when running tasks in the subprojects, which should only be in the `each` task.